src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -1022,28 +1028,11 @@ bool clang_c_convertert::get_type(const clang::Type &the_type, typet &new_type)
     std::string id, name;
     get_decl_name(rd, name, id);
 
-    // does symbol already exist before we go through the getter?
-    bool sym_already_exist = context.find_symbol(id);
-
-    // avoid unnecessary conversion if symbol already exists
-    if(!sym_already_exist)
-      if(get_struct_union_class(rd))
-        return true;
-
-    symbolt &s = *context.find_symbol(id);
-    new_type = s.type;
+    /* record in context if not already there */
+    get_struct_union_class(rd);
```

Comment:
Do we need this here?  `get_type` is only called when we try to get the type of an expr, which either has its type already defined or is being defined (like the C++ iterator example from @kunjsong01).

Can't we just the ` new_type = symbol_typet(id);`?

Reply:
That would lead to the definition of the anonymous union in `__mbstate_t` missing, see <https://github.com/esbmc/esbmc/pull/1180#issuecomment-1640037798>.